From 9b7f3d4920e85c3e7befee31aceef81c473c4ebc Mon Sep 17 00:00:00 2001
From: John Zimmermann <me@johnnynator.dev>
Date: Tue, 6 Apr 2021 00:30:54 +0200
Subject: [PATCH] Use lld instead of Ld as format specifier

The C Standard uses L for conversion to long doubles,
ll should be used for conversion to long long int's

BUG: 418142
---
 processcore/processes_linux_p.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/processcore/processes_linux_p.cpp b/processcore/processes_linux_p.cpp
index adbcadd..3ed1271 100644
--- a/processcore/processes_linux_p.cpp
+++ b/processcore/processes_linux_p.cpp
@@ -161,7 +161,7 @@ bool ProcessesLocal::Private::readProcStatus(const QString &dir, Process *proces
             qlonglong euid;
             qlonglong suid;
             qlonglong fsuid;
-            sscanf(mBuffer + sizeof("Uid:") -1, "%Ld %Ld %Ld %Ld", &uid, &euid, &suid, &fsuid);
+            sscanf(mBuffer + sizeof("Uid:") -1, "%lld %lld %lld %lld", &uid, &euid, &suid, &fsuid);
             process->setUid(uid);
             process->setEuid(euid);
             process->setSuid(suid);
@@ -172,7 +172,7 @@ bool ProcessesLocal::Private::readProcStatus(const QString &dir, Process *proces
 	  case 'G':
 	    if((unsigned int)size > sizeof("Gid:") && qstrncmp(mBuffer, "Gid:", sizeof("Gid:")-1) == 0) {
             qlonglong gid, egid, sgid, fsgid;
-            sscanf(mBuffer + sizeof("Gid:")-1, "%Ld %Ld %Ld %Ld", &gid, &egid, &sgid, &fsgid);
+            sscanf(mBuffer + sizeof("Gid:")-1, "%lld %lld %lld %lld", &gid, &egid, &sgid, &fsgid);
             process->setGid(gid);
             process->setEgid(egid);
             process->setSgid(sgid);
-- 
GitLab

